### PR TITLE
chore: update WebdriverIO packages

### PR DESCRIPTION
**Pull request by christian-bromann**
#### Prerequisites checklist

- [x] I have read the [contributing guidelines](https://github.com/eslint/eslint/blob/HEAD/CONTRIBUTING.md).

#### What is the purpose of this pull request? (put an "X" next to an item)

[ ] Documentation update
[x] Bug fix ([template](https://raw.githubusercontent.com/eslint/eslint/HEAD/templates/bug-report.md))
[ ] New rule ([template](https://raw.githubusercontent.com/eslint/eslint/HEAD/templates/rule-proposal.md))
[ ] Changes an existing rule ([template](https://raw.githubusercontent.com/eslint/eslint/HEAD/templates/rule-change-proposal.md))
[ ] Add autofix to a rule
[ ] Add a CLI option
[ ] Add something to the core
[ ] Other, please explain:

Fixes https://github.com/eslint/eslint/issues/18548

#### What changes did you make? (Give an overview)

I update the WebdriverIO packages for browser tests to pass again more reliable. WebdriverIO has released [v8.38.3](https://github.com/webdriverio/webdriverio/releases/tag/v8.38.3) which fixes some performance issues we've discovered with some changes introduced to the test harness in Eslint.

### CODE REVIEW

File: package.json [hunk 1]
```diff
@@ -107,11 +107,11 @@
...
-    "@wdio/browser-runner": "8.36.0",
-    "@wdio/cli": "^8.14.6",
-    "@wdio/concise-reporter": "^8.14.0",
-    "@wdio/globals": "^8.14.6",
-    "@wdio/mocha-framework": "^8.14.0",
+    "@wdio/browser-runner": "^8.38.3",
+    "@wdio/cli": "^8.38.2",
+    "@wdio/concise-reporter": "^8.38.2",
+    "@wdio/globals": "^8.38.2",
+    "@wdio/mocha-framework": "^8.38.2",
```
Comment (snitin315) [inline package.json:114]: Thanks for the update, but the browser tests are still failing. [Reference](https://github.com/eslint/eslint/actions/runs/9377636509/job/25819524336?pr=18558)

**Comment by aladdin-add** [comment]
it did not pass: https://github.com/eslint/eslint/actions/runs/9377636509/job/25819524336?pr=18558

**Comment by nzakas** [comment]
@christian-bromann it looks like this is still failing, can you take a look?

**Comment by christian-bromann** [comment]
@nzakas @aladdin-add not sure why it was failing but I don't see them failing anymore.

**Review by mdjermanovic — APPROVED** [review]
Okay, let's try this since the tests often fail with pinned 8.36.0 anyway.

**Comment by christian-bromann** [comment]
Please keep me posted if failures pop up again.

**Pull request merged**